api/admin_test.go

```diff
@@ -0,0 +1,211 @@
+	}
+
+	u, err = models.NewUser(email, "test", ts.API.config.JWT.Aud, nil)
+	require.NoError(ts.T(), err, "Error making new user")
+
+	u.IsSuperAdmin = true
+	require.NoError(ts.T(), ts.API.db.CreateUser(u), "Error creating user")
+
+	token, err := ts.API.generateAccessToken(u)
+	require.NoError(ts.T(), err, "Error generating access token")
+
+	tok, err := jwt.Parse(token, func(token *jwt.Token) (interface{}, error) {
+		assert.Equal(ts.T(), token.Header["alg"], "HS256")
+		return []byte(ts.API.config.JWT.Secret), nil
+	})
+	require.NoError(ts.T(), err, "Error parsing token")
+
+	// Setup response recorder
+	w := httptest.NewRecorder()
+	ctx := req.Context()
+
+	return context.WithValue(ctx, "jwt", tok), w
+}
+
+// TestAdminUsers tests API /admin/users route
+func (ts *AdminTestSuite) TestAdminUsers() {
+	// Setup request
+	req := httptest.NewRequest("GET", "/admin/users", nil)
+
+	// Setup response recorder with super admin privileges
+	ctx, w := ts.makeSuperAdmin(req, "test@example.com")
+
+	ts.API.adminUsers(ctx, w, req)
+
+	resp := w.Result()
+
+	data := make(map[string]interface{})
+	require.NoError(ts.T(), json.NewDecoder(resp.Body).Decode(&data))
+
+	assert.Equal(ts.T(), resp.StatusCode, 200)
+	for _, user := range data["users"].([]interface{}) {
+		assert.NotEmpty(ts.T(), user)
+	}
+}
+
+// TestAdminUserCreate tests API /admin/user route (POST)
+func (ts *AdminTestSuite) TestAdminUserCreate() {
+	var buffer bytes.Buffer
+	require.NoError(ts.T(), json.NewEncoder(&buffer).Encode(map[string]interface{}{
+		"email":    "test1@example.com",
+		"password": "test1",
+	}))
+
+	// Setup request
+	req := httptest.NewRequest("POST", "/admin/user", &buffer)
+
+	// Setup response recorder with super admin privileges
+	ctx, w := ts.makeSuperAdmin(req, "test@example.com")
+
+	ts.API.adminUserCreate(ctx, w, req)
+
+	resp := w.Result()
+	assert.Equal(ts.T(), resp.StatusCode, 200)
+
+	u, err := ts.API.db.FindUserByEmailAndAudience("test1@example.com", ts.API.config.JWT.Aud)
+	require.NoError(ts.T(), err)
+
+	data := make(map[string]interface{})
+	require.NoError(ts.T(), json.NewDecoder(resp.Body).Decode(&data))
+
+	assert.Equal(ts.T(), data["email"], u.Email)
+}
+
+// TestAdminUserGet tests API /admin/user route (GET)
+func (ts *AdminTestSuite) TestAdminUserGet() {
+	var buffer bytes.Buffer
+	require.NoError(ts.T(), json.NewEncoder(&buffer).Encode(map[string]interface{}{
+		"user": map[string]interface{}{
+			"email": "test1@example.com",
+			"aud":   ts.API.config.JWT.Aud,
+		},
+	}))
+
+	u, err := models.NewUser("test1@example.com", "test", ts.API.config.JWT.Aud, nil)
+	require.NoError(ts.T(), err, "Error making new user")
+	require.NoError(ts.T(), ts.API.db.CreateUser(u), "Error creating user")
+
+	// Setup request
+	req := httptest.NewRequest("GET", "/admin/user", &buffer)
+
+	// Setup response recorder with super admin privileges
+	ctx, w := ts.makeSuperAdmin(req, "test@example.com")
+
+	ts.API.adminUserGet(ctx, w, req)
+
+	resp := w.Result()
```

Comment:
instead of pulling the result of the ResponseRecorder you can directly use its buffer and return codes.